source/multigrid/mg_transfer_internal.cc

```diff
@@ -131,42 +143,67 @@ namespace internal
                         level, level_dof_indices[i]))
                     continue;
 
-                  types::global_dof_index global_idx =
-                    globally_relevant.index_within_set(global_dof_indices[i]);
-                  // skip if we did this global dof already (on this or a
-                  // coarser level)
-                  if (dof_touched[global_idx])
-                    continue;
```

Comment:
Why do we need to move this?

Reply:
It is usually much cheaper to check the two cases `global_mine` and `level_mine` which go to the locally owned sets, which in most cases consists of a single range or at most a few only. The locally relevant dofs, on the other hand, contain many ranges due to the ghost layer. Therefore I decided to only do it in case we need to mark the indices.

Reply:
I will soon push the code with a comment about this fact.